api/recommend_func.js

```diff
@@ -0,0 +1,79 @@
+const recombee = require('recombee-api-client');
+
+const rqs = recombee.requests;
+const recombeeClient = new recombee.ApiClient(process.env.RECOMBEE_DATABASE_ID,
+  process.env.RECOMBEE_PRIVATE_TOKEN);
+
+async function recPersonal(userHash, count) {
+  let result;
+  await recombeeClient.send(
+    new rqs.RecommendItemsToUser(userHash, count, { scenario: 'personal_view' }),
+  )
+    .then((recommendation) => {
+      result = { recipes: recommendation.recomms.map((e) => e.id) || [] };
+    })
+    .catch(() => {
+      recombeeClient.send(
+        new recombee.RecommendItemsToUser(userHash, count, { scenario: 'homepage_view' }),
+      );
+    })
+    .then((recommendation) => {
+      result = { recipes: recommendation.recomms.map((e) => e.id) || [] };
+    })
+    .catch((err) => {
+      throw (err);
+    });
+  return result;
+}
+
+async function recPopular(userHash, count) {
+  let result;
+  await recombeeClient.send(
+    new rqs.RecommendItemsToUser(userHash, count, { scenario: 'popular_view' }),
+  )
+    .then((recommendation) => {
+      result = { recipes: recommendation.recomms.map((e) => e.id) || [] };
+    })
+    .catch(() => { result = { recipes: [] }; })
+    .catch((err) => {
+      throw (err);
+    });
+  return result;
+}
+
+
+async function recommendFunc(userHash, count, type, isAnonymous) {
```

Comment:
this is a bad interface, very adhoc to your needs. Also why have isAnonymous, just pass a hash or don't depending on if you have it or not

Reply:
you don't need a giant switch case, this is a bad design, you need to export the separate functions and let everyone use them.